apps/andi/lib/andi/input_schemas/input_converter.ex

```diff
@@ -266,7 +268,22 @@ defmodule Andi.InputSchemas.InputConverter do
   defp update_context_from_smrt_step(context, _), do: context
 
   defp encode_extract_step_body_as_json(%{body: body} = smrt_extract_step) when body not in ["", nil] do
-    Map.put(smrt_extract_step, :body, Jason.encode!(body))
+    case body do
+      _ when is_binary(body) ->
+        raise_error_if_invalid_json(body)
+        Map.put(smrt_extract_step, :body, body)
+
+      _ when is_map(body) ->
+        Map.put(smrt_extract_step, :body, Jason.encode!(body))
+
+      _ ->
+        Logger.error("Received an extract step body that is not a string or a map. Received body: #{body}}")
+        smrt_extract_step
```

Comment:
Looks great. Plus if this very hard to reach error happens, reaper will still attempt to make the request without the body, which should provide clear logs in the event the server rejects due to a missing body. Instead of reaper just never attempting.